image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,124 @@
+  - "@bparees"
+  - "@sdodson"
+  - "@rgolangh" # RHV
+  - "@russellb" # k8s-native infra
+approvers:
+  - "@bparees"
+  - "@smarterclayton"
+  - "@derekwaynecarr"
+creation-date: 2019-09-25
+last-updated: 2019-09-25
+status: provisional
+see-also: 
+replaces:
+superseded-by:
+---
+
+# Remove Registry on Baremetal
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+On `baremetal` platform clusters [link], the OpenShift image registry operator will bootstrap itself as `Removed`.
+This will allow `openshift-installer` to complete installations on this platform type.
+A separate process will be responsible for configuring PVC-compatible storage [link] and updating the image registry operator's state to `Managed` [link].
+
+## Motivation
+
+IPI installation options for Kubernetes-native infrastructure (formerly RHII.next) and Red Hat Virtualization (RHV) have been declared a key initiative for OpenShift 4.3.
+These will configure themselves as the `baremetal` platform type.
+
+### Goals
+
+- `openshift-installer` completes if the cloud provider is set to `baremetal`.
+- Image registry is able to bring itself up after the management state is switched from `Removed` to `Managed`.
+
+### Non-Goals
+
+- Provision storage for the registry for RHV and k8s-native infrastructre.
+- Provide 1st class support for additional storage types/providers for the registry (ex: MCG).
+- Harden OpenShift features (ex: Builds, ImageStreams) such that they error cleanly if the registry is not installed.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Story
+
+As an OpenShift cluster admin/installer
+I want the registry to be `Removed` on RHV and k8s-native infra installations
+So that I can provision other storage providers
+And later enable the image registry with the desired storage
+
+### Implementation Details/Notes/Constraints
+
+- When the image registry operator bootstraps itself, it will inspect the cluster infrastructure platform status.
+- If the platform status type is `baremetal`, the registry will mark itself `Removed`.
+- A separate installation process will be resposible for provisioning storage. For RHV and k8s-native infra, this is expected to be a PVC with the `RWX` access mode.
+
+### Risks and Mitigations
+
+#### Image registry is never installed
+
+Lots of components will likely report errors if a customer runs the installer as `baremetal` and does nothing to set up the registry.
+Potential candidates that will break:
+1. Builds which push to an imagestream tag - will error at runtime.
+2. Samples operator - reports degraded if imagestreams fail to import.
+3. Imagestream import with `--reference-policy=local` may always fail.
```

Comment:
i'm not sure if the import itself will fail here.

the real issue is that an imagestream using local ref policy (including all the ones the samples operator creates) will be unusable by components that respect the local ref policy (ie resolve the imagestreamtag pullspec by pointing to the internal registry, not the upstream registry).  Which of course includes any buildconfig that points to one of these imagestreamtags, as well as DeploymentConfigs.

actually they may not be unusable(in the sense of not resolving properly).....there will be no internal registry hostname, so the imagestream won't list one (I assume).  So the resolved pullspec should fallback to the external registry.  What will then fail, of course, is the pull will fail because it has no auth for the upstream registry, unless the user explicitly adds.